### PR TITLE
docs(bigquery): remove location parameter from samples

### PR DESCRIPTION
**Pull request by plamut**
Fixes [#9499](https://github.com/googleapis/google-cloud-python/issues/9499).
Supersedes [#9500](https://github.com/googleapis/google-cloud-python/pull/9500).

Following the [comment](https://github.com/googleapis/google-cloud-python/pull/9500#issuecomment-544137072) on the original PR, I am submitting the changes directly to a related PR to avoid merge conflicts when merging all the work into the main Google Cloud Python repository.

### CODE REVIEW

**Comment by plamut** [comment]
@Emar-Kar I left the `location` argument in a few places in the samples where I thought we want to keep it to demonstrate its use cases (e.g. explicitly setting the location of a new dataset).

**Comment by plamut** [comment]
@Emar-Kar Any chance of merging this? That way we avoid potential merge conflicts when you guys make any changes to the `new-samples` branch.

Thanks! :)

**Comment by Emar-Kar** [comment]
@plamut I think you need to ask Tim Swast for the review first. He should confirm that the `location` parameter should be deleted not only at the `query` methods. And I assume he will ask to merge this PR directly into `google-cloud-python` repo, instead of double merge through another PR.

**Comment by plamut** [comment]
Since my set of changes is more trivial, I'm fine with waiting until the `new-samples` branch is merged to the main repo. Will be easier to resolve any further merge conflicts on my side.

**Comment by tswast** [comment]
I'm fine reviewing this PR here now that I found it in my notifications. :-)

File: bigquery/samples/client_query_batch.py [hunk 1]
```diff
@@ -32,13 +32,11 @@ def client_query_batch(client):
...
     # Start the query, passing in the extra configuration.
-    query_job = client.query(
-        sql, location="US", job_config=job_config
-    )  # Make an API request.
+    query_job = client.query(sql, job_config=job_config)  # Make an API request.
 
     # Check on the progress by getting the job's updated state. Once the state
     # is `DONE`, the results are ready.
-    query_job = client.get_job(query_job.job_id, location="US")  # Make an API request.
+    query_job = client.get_job(query_job.job_id)  # Make an API request.
```
Comment (tswast) [inline bigquery/samples/client_query_batch.py]: Actually, I don't believe `get_job` does any auto-detection except for defaulting to `US`. Luckily, we can extract the location from the job resource:

```suggestion
    query_job = client.get_job(query_job.job_id, location=query_job.location)  # Make an API request.
```

File: bigquery/samples/tests/test_create_job.py [hunk 2]
```diff
@@ -19,6 +19,6 @@
 def test_create_job(capsys, client):
 
     query_job = create_job.create_job(client)
-    client.cancel_job(query_job.job_id, location="US")
+    client.cancel_job(query_job.job_id)
```
Comment (tswast) [inline bigquery/samples/tests/test_create_job.py]: Same here. I don't believe the cancel operation supports auto-detection. It defaults to `US`.

```suggestion
    client.cancel_job(query_job.job_id, location=query_job.location)
```

**Pull request merged**